src/yaml/serialization.cr

```diff
@@ -125,6 +125,27 @@ module YAML
   # field, and the rest of the fields, and their meaning, depend on its value.
   #
   # You can use `YAML::Serializable.use_yaml_discriminator` for this use case.
+  #
+  # ### `after_initialize` method
+  #
+  # after_initialize is a method that runs after initialization and can be used
+  # as a hook to post-process the initialized object.
+  #
+  # Example:
+  # ```
+  # require "yaml"
+  #
+  # class Person
+  #   include YAML::Serializable
+  #   @name : String
+  #
+  #   def after_initialize
+  #     @name = @name.upcase
+  #   end
+  # end
+  #
+  # pp Person.from_yaml("---\nname: Jane\n") # => #<Person:0x7f630cb92e80 @name="JANE">
```

Comment:
The same applies here:
```suggestion
  # person = Person.from_yaml("---\nname: Jane\n")
  # person.name # => "JANE"
```

Reply:
Now that we access the `name` variable in the example, I should make it available. I could do the following to have a continuity with the previous examples:

```cr
require "json"
  
class Person
  include JSON::Serializable

     @[JSON::Field(key: "name")]
     property name : String
  
   def after_initialize
     @name = @name.upcase
   end
end
 
person = Person.from_json(%({"name": "Jane"}))
person.name # => "JANE"
```

Reply:
There's no need for the `JSON::Field` annotation. Just replacing `@name : String` with `getter name : String` should be sufficient.